cbqueryx/queryrespreader.go

```diff
@@ -122,27 +125,28 @@ func (r *queryRespReader) parseError(errJson *queryErrorJson) *ServerError {
 	if errCodeGroup == 5 {
 		err = ErrInternalServerError
 	}
-	if errCodeGroup == 12 || errCodeGroup == 14 && errCode != 12004 && errCode != 12016 {
```

Comment:
Do we need to add a new if statement for  `12004` or is this alright to be `ErrIndexFailure` now?

Reply:
You're correct, although the existing code was actually already wrong, so I've filed an independent issue to address the fact that our 12004 cut-out wasn't actually working as it was likely intended to: https://couchbasecloud.atlassian.net/browse/ING-655